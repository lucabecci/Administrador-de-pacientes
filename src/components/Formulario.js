import React, { useState } from 'react'
import uuid from 'uuid/dist/v4'
import PropTypes from 'prop-types'

const Formulario = ({crearCita}) => {

    //Crear State De Citas
    const [cita, actualizarCita] = useState({
        mascota: '',
        propietario: '',
        fecha: '',
        hora: '',
        sintomas: ''           
    })

    const [error, actualizarError] = useState(false);
    //Funcion que se ejecuta cada vez que se escribe en un input
    const actualizarState = (e) =>{
        actualizarCita({
            ...cita,
            [e.target.name]: e.target.value
        })
    }
    //Ectraer los valores
    const { mascota, propietario, fecha, hora, sintomas } = cita;
    //Cuando el usuario presiona enviar cita
    const submitCita = e =>{
        e.preventDefault()
        //validar
        if(mascota.trim() === '' || propietario.trim() === '' || fecha.trim() === '' || 
        hora.trim() === '' || sintomas.trim() === ''){
                actualizarError(true)
                return;
            }

        //eliminar el mensaje previo
        actualizarError(false)    
        //asignar un ID
            cita.id = uuid();
        //crear la cita
            crearCita(cita)
        //resetear el form
        actualizarCita({
            mascota: '',
            propietario: '',
            fecha: '',
            hora: '',
            sintomas: ''  
        })
    }

    return (
        <>   
        <h2>Crear Cita</h2>
        { error ? <p className = 'alerta-error'>TODOS LOS CAMPOS DEBEN DE ESTAR LLENOS</p>  : null}
        <form
            onSubmit = {submitCita}
        >
            <label>Nombre De La Mascota:</label>
            <input
                type = "text"
                name = "mascota"
                className = "u-full-width"
                placeholder = "Nombre De La Mascota "
                onChange = {actualizarState}
                value ={mascota}
            />
            <label>Nombre Del Dueño:</label>
            <input
                type = "text"
                name = "propietario"
                className = "u-full-width"
                placeholder = "Nombre Del Dueño"
                onChange = {actualizarState}
                value ={propietario}
            />
            <label>Fecha:</label>
            <input
                type = "date"
                name = "fecha"
                className = "u-full-width"
                onChange = {actualizarState}
                value ={fecha}
            />
            <label>Hora:</label>
            <input
                type = "time"
                name = "hora"
                className = "u-full-width"
                onChange = {actualizarState}
                value = {hora}
            />
            <label>Sintomas:</label>
            <textarea
                className = 'u-full-width'
                name = 'sintomas'
                onChange = {actualizarState}
                value = {sintomas}
            ></textarea>

            <button
                type = 'submit'
                className = 'u-full-width button-primary'
            >
                Agregar Cita.
            </button>
        </form>
        </>
    )
}
//PROPTYPES
Formulario.propTypes ={
    crearCita: PropTypes.func.isRequired
}

export default Formulario
