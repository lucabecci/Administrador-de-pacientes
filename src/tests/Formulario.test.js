import React from 'react'
import Formulario from '../components/Formulario'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

describe('Test en Formulario', () => {

    let wrapper = shallow(<Formulario/>)

    test('Snapshot en Formulario', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
})
