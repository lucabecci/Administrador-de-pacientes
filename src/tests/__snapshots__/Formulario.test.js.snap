// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en Formulario Snapshot en Formulario 1`] = `
<Fragment>
  <h2>
    Crear Cita
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label>
      Nombre De La Mascota:
    </label>
    <input
      className="u-full-width"
      name="mascota"
      onChange={[Function]}
      placeholder="Nombre De La Mascota "
      type="text"
      value=""
    />
    <label>
      Nombre Del Dueño:
    </label>
    <input
      className="u-full-width"
      name="propietario"
      onChange={[Function]}
      placeholder="Nombre Del Dueño"
      type="text"
      value=""
    />
    <label>
      Fecha:
    </label>
    <input
      className="u-full-width"
      name="fecha"
      onChange={[Function]}
      type="date"
      value=""
    />
    <label>
      Hora:
    </label>
    <input
      className="u-full-width"
      name="hora"
      onChange={[Function]}
      type="time"
      value=""
    />
    <label>
      Sintomas:
    </label>
    <textarea
      className="u-full-width"
      name="sintomas"
      onChange={[Function]}
      value=""
    />
    <button
      className="u-full-width button-primary"
      type="submit"
    >
      Agregar Cita.
    </button>
  </form>
</Fragment>
`;
